app/src/main/java/org/fossasia/openevent/general/speakers/SpeakerWithEventDao.kt

```diff
@@ -0,0 +1,23 @@
+package org.fossasia.openevent.general.speakers
+
+import androidx.lifecycle.LiveData
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import androidx.room.RoomWarnings
+
+@Dao
+interface SpeakerWithEventDao {
+    @SuppressWarnings(RoomWarnings.CURSOR_MISMATCH)
+    @Query("""
+        SELECT * FROM speaker
+        INNER JOIN speakerwithevent ON
+        speaker.id = speakerwithevent.speaker_id
+        WHERE speakerwithevent.event_id = :eventID
+        """)
+    fun getSpeakerWithEventId(eventID: Long): LiveData<List<Speaker>>
```

Comment:
This should be in SpeakerDao

Reply:
SpeakerWithEvent Dao maps m:n relationship of speakers and events

Reply:
@iamareebjamal there is a seperate class for SpeakerDao

Reply:
Still, why can't this be in SpeakerDao?

Reply:
If I'll be passing the id of event ,speaker table will not recognize it as there is no column for eventId